src/atoms/SegmentControl/__tests__/SegmentControl.native.test.js

```diff
@@ -0,0 +1,478 @@
+        </SegmentControl>,
+      );
+      const option = getByTestId('segment-option-2');
+      fireEvent.press(option);
+      expect(mockPress).toBeCalledWith('2');
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('size', () => {
+    it('renders small outlined segment control ', () => {
+      const { container } = renderWithTheme(
+        <SegmentControl defaultValue="1" variant="outlined" size="small">
+          <SegmentControl.Option value="1" testID="segment-option-1">
+            Option 1
+          </SegmentControl.Option>
+          <SegmentControl.Option value="2" testID="segment-option-2">
+            Option 2
+          </SegmentControl.Option>
+          <SegmentControl.Option value="3" testID="segment-option-3">
+            Option 3
+          </SegmentControl.Option>
+        </SegmentControl>,
+      );
+      expect(container).toMatchSnapshot();
+    });
+    it('renders medium outlined segment control ', () => {
+      const { container } = renderWithTheme(
+        <SegmentControl defaultValue="1" variant="outlined" size="medium">
+          <SegmentControl.Option value="1" testID="segment-option-1">
+            Option 1
+          </SegmentControl.Option>
+          <SegmentControl.Option value="2" testID="segment-option-2">
+            Option 2
+          </SegmentControl.Option>
+          <SegmentControl.Option value="3" testID="segment-option-3">
+            Option 3
+          </SegmentControl.Option>
+        </SegmentControl>,
+      );
+      expect(container).toMatchSnapshot();
+    });
+
+    it('renders small filled segment control ', () => {
+      const { container } = renderWithTheme(
+        <SegmentControl defaultValue="1" variant="filled" size="small">
+          <SegmentControl.Option value="1" testID="segment-option-1">
+            Option 1
+          </SegmentControl.Option>
+          <SegmentControl.Option value="2" testID="segment-option-2">
+            Option 2
+          </SegmentControl.Option>
+          <SegmentControl.Option value="3" testID="segment-option-3">
+            Option 3
+          </SegmentControl.Option>
+        </SegmentControl>,
+      );
+      expect(container).toMatchSnapshot();
+    });
+
+    it('renders medium filled segment control ', () => {
+      const { container } = renderWithTheme(
+        <SegmentControl defaultValue="1" variant="filled" size="medium">
+          <SegmentControl.Option value="1" testID="segment-option-1">
+            Option 1
+          </SegmentControl.Option>
+          <SegmentControl.Option value="2" testID="segment-option-2">
+            Option 2
+          </SegmentControl.Option>
+          <SegmentControl.Option value="3" testID="segment-option-3">
+            Option 3
+          </SegmentControl.Option>
+        </SegmentControl>,
+      );
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('pressIn', () => {
+    it('outlined segment control unselected', () => {
```

Comment:
shouldn't unselected be without `defaultValue`? I guess that's what the aim is?

Reply:
This is testing pressIn behaviour for unselected segment. It has a different underlay color that a selected segment